app/src/main/java/edu/uwstout/p2pchat/SettingNode.java

```diff
@@ -0,0 +1,387 @@
+package edu.uwstout.p2pchat;
+
+import android.app.Activity;
+import android.content.Context;
+import android.graphics.Color;
+import android.util.Log;
+import android.view.View;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentManager;
+import androidx.fragment.app.FragmentTransaction;
+import androidx.navigation.NavController;
+import androidx.navigation.NavDirections;
+import androidx.navigation.NavGraph;
+import androidx.navigation.NavHostController;
+import androidx.navigation.Navigation;
+import androidx.navigation.fragment.NavHostFragment;
+import androidx.navigation.ui.NavigationUI;
+
+import java.util.LinkedList;
+
+
+/**
+ * This class here is responsible putting messy code into
```

Comment:
documentation isn't a clear description of what it does

Reply:
this class also has a lot of really bad code. If I added even one more child view, the refactoring would be a nightmare. REDO the whole class or get rid of it. Use lists or arrays or some data structure instead of an explicit variable called childView1, childView2, etc.